### ISSUE TITLE
Game interface design

### ISSUE THREAD

**Issue by ExcaliburZero** [Infinite-Dango-Studios/ganbatte#8]
Currently the game is set to use a default interface theme, however, while this is okay for now, we will eventually need to change it to something more customized for the game.

I can probably handle most of the coding for the interface, but I will need the help of the artists to come up with what the style of the interface will be.

Essentially what we need is a style for windows and general screens. So, at least a color scheme and some sort of background texture for text windows.

This isn't currently a very important issue, but it is something that will eventually need to be addressed.

**Comment by boxesofflowers** [comment]
I think starting a color palette for the game would be a really good idea, to keep everything consistent. Hex codes and everything, so it's not just guess-and-check.

**Comment by ExcaliburZero** [comment]
@boxesofflowers 
That sounds good. It would be a good idea to come up with a color palette beforehand to keep everything consistent.

**Comment by ExcaliburZero** [comment]
Does anyone have any ideas for what the color theming for the game should be?

**Comment by juanitaylor** [comment]
I feel like warm colors, browns, oranges, etc, would be good, since our theme is cafe?

**Comment by boxesofflowers** [comment]
I agree with Jacoba, warm colors would be good, especially darker warm colors, because I feel like the story is sort of...darkish? it would be cool if the color palette changed depending on the path you choose, but that might be a little cray

**Comment by Sabre-Hecate** [comment]
Not brown AND orange. Anything but those two together. I don't know why, but I sort of don't like those two together? IDK...maybe I'll like what you two do with it.

**Comment by ExcaliburZero** [comment]
Warm colors sound good.

@boxesofflowers 
I'm not sure if it would be possible to change the color pallet in different points of the game, but I might try looking into that at some point to see if it is possible.

**Comment by ExcaliburZero** [comment]
I've started doing so testing of the customization features of the engine. You can try the latest version to see how it currently looks. Here is a screenshot of what it looks like currently:

![sc - 07](https://cloud.githubusercontent.com/assets/4276196/6475567/c083b598-c1d9-11e4-9f47-0fb86f427e5c.png)

We can customize just about everything without too much difficulty. I also tested out making box border styles with images, with an example being the text box in the screenshot made with this source image:

![box_border](https://cloud.githubusercontent.com/assets/4276196/6475589/03257454-c1da-11e4-967f-b1332ced9b54.png)

Let me know what you come up with for the color scheme and I'll work to incorporate it into the game.

**Comment by Sabre-Hecate** [comment]
Looks nice so far!

**Comment by ExcaliburZero** [comment]
I also just tested out character labels, which can put the character names in a separate box:

![sc - 08](https://cloud.githubusercontent.com/assets/4276196/6476582/dc965298-c1e5-11e4-9607-63352cba09fa.png)

**Comment by Sabre-Hecate** [comment]
I think maybe I like # 2 better. _shrug_

**Comment by Sabre-Hecate** [comment]
Didn't mean to put that link there...

**Comment by juanitaylor** [comment]
Changing the color pallet might be weird, and distract from the game, but if it's subtle, it might be effective _le shrug_

**Comment by ExcaliburZero** [comment]
I have implemented separate character name boxes that I screenshotted earlier into the game with 3740715c0cf381677e407f96cdec154dc66af038 . We can go with this system or we can go with the original system depending on what we come up with for the design.

Another example:

![sc - 09](https://cloud.githubusercontent.com/assets/4276196/6535542/b3d06d14-c413-11e4-9307-f69b6855cffe.png)

**Comment by ExcaliburZero** [comment]
Also, a slightly interface related issue is text speed. Currently when you click on the screen all of the text of the next dialogue comes up at once, however the text speed can also be set up to display the letters over a second or so, like a typewriter effect. If we want to use that we can set a default speed and the player can modify that speed if they want. Or we can leave the feature disabled as it is now. I'm not sure which option would work better.

**Comment by juanitaylor** [comment]
definitely not all at once. it's cool if it appears like a typewriter owo

**Comment by ExcaliburZero** [comment]
@juanitaylor 
Yeah, I think a slight typewriter effect might work better. But we should make sure the timing of it flows well.

I'll add in the typewriter effect, and try working around with the settings a bit. Worst comes to worst, we can always just switch it back.

I recommended you try testing it out once I finish playing with the settings a bit.

**Comment by ExcaliburZero** [comment]
I added a typewritter (text speed) effect and a click to continue icon in 38a117d5901a7f337c9414495e2d98545e4d7790 . Test it out and let me know if it needs any adjusting.

**Comment by ExcaliburZero** [comment]
We should probably try putting together some sort of color pallet for the game, that way we can iron out the color scheme a bit.

**Comment by boxesofflowers** [comment]
I'll work on color palette. As for other interface-type things, font is going to change and backgrounds are going to be drawn. I checked out the sample, and the mechanics are nice, the only thing is there's a lot of redundant text, (since it's a visual novel, less detailed descriptions will work) and since the game is long, a buttload of exposition isn't really necessary, because the player will ideally learn this information as it pertains to the plot later on. also, I was playing for ~10 minutes and there were only two opportunities to make a choice. ideally, every time you're replying to something someone's said, there should be a choice. it doesn't have to change the path, it just engages the player and may change the next response. 
ex. 
Character A: Let's go!
Main Character: Sure.
vs.
Character A: Let's go!
Main Character: (Sure!) (Ugh, ok.) (Where?)

**Comment by ExcaliburZero** [comment]
@boxesofflowers 
Let me know what progress you make on the color pallet, so I can try to incorporate it into the website.

With fonts we'll probably have to find a font that comes with most operating systems by default, or ship the font file with the project.

As for the game dialogue choices, Sabre-Hecate has been going with a style similar to Katawa Shoujo, whereby there are less choices for the player to make, but each choice fuels the dialogue that follows it and thus each choice has more meaning. But that's a bit off topic for this issue.

**Comment by ExcaliburZero** [comment]
@boxesofflowers 
Once you finish making the color pallet you should probably add it to the [Ganbatte resources repo](https://github.com/Infinite-Dango-Studios/ganbatte-resources). Let me know if you need any help doing so.

**Comment by ExcaliburZero** [comment]
Since boxesofflowers has offered to work on the color pallet, I'll assign this issue to her for now.
